### PR TITLE
[CI][runtime-env] fix runtime-env test

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

This PR adds 3.11 in the allowed runtime env conda version which was made available during the recent conda 2.23.0 release. https://github.com/conda/conda/issues/11170#issuecomment-1466134840

It also fixes a a wheel rename for mac osx wheels from "*_intel.whl" to "*x86_64.whl" in recent builds. 
- A previous intel wheel build for mac:https://buildkite.com/ray-project/oss-ci-build-branch/builds/1268#0184a81c-6dd7-4cd2-8534-b6f763b8ce24
- A more recent x86_64 name wheel build:  https://buildkite.com/ray-project/oss-ci-build-branch/builds/2599#0186badf-4b95-44fd-b57d-6e42d5462c1d

## Related issue number
TODO: Need to make sure minimal test runs in the expected conda version: 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by scv119 — APPROVED** [review]
thanks! looks good to unblock the CI.

File: python/ray/_private/utils.py [hunk 1]
```diff
@@ -1271,9 +1271,7 @@ def get_wheel_filename(
     assert py_version in ray_constants.RUNTIME_ENV_CONDA_PY_VERSIONS, py_version
 
     py_version_str = "".join(map(str, py_version))
-    if py_version_str in ["36", "37"]:
```
Comment (rickyyx) [inline python/ray/_private/utils.py:1274]: This was changed. Any implications of this? @architkulkarni

Comment (rickyyx) [reply python/ray/_private/utils.py:1274]: Alternatively, we could keep this - and I will basically skip all the py-311 wheel checks

**Comment by architkulkarni** [comment]
@rickyyx Good catch on the name change from `intel` to `x86-64` on 3.6 and 3.7.  That means the `runtime_env` `"conda"` field won't work on 3.6 or 3.7 for in Ray 3.1 or on the nightly wheels without this fix.  We should 
1. Cherry pick this for the release if possible, and 
2. Find a way to run some kind of test in CI on all Python versions to catch this kind of name change before it happens (You noticed the name change manually and not because of some test failure, right?)

File: python/ray/_private/utils.py (hunk 1 above)
Comment (architkulkarni) [reply python/ray/_private/utils.py:1274]: No other implications outside of this code itself.  I think some others are out of date and should be fixed too, for example it looks like `universal` is no longer part of the name of any of the wheels:

This is the list of wheels from the 2.3.1 branch
```
ray-2.3.1-cp310-cp310-manylinux2014_aarch64.whl
ray-2.3.1-cp310-cp310-manylinux2014_x86_64.whl
ray-2.3.1-cp310-cp310-win_amd64.whl
ray-2.3.1-cp311-cp311-manylinux2014_aarch64.whl
ray-2.3.1-cp311-cp311-manylinux2014_x86_64.whl
ray-2.3.1-cp36-cp36m-manylinux2014_aarch64.whl
ray-2.3.1-cp36-cp36m-manylinux2014_x86_64.whl
ray-2.3.1-cp37-cp37m-manylinux2014_aarch64.whl
ray-2.3.1-cp37-cp37m-manylinux2014_x86_64.whl
ray-2.3.1-cp37-cp37m-win_amd64.whl
ray-2.3.1-cp38-cp38-manylinux2014_aarch64.whl
ray-2.3.1-cp38-cp38-manylinux2014_x86_64.whl
ray-2.3.1-cp38-cp38-win_amd64.whl
ray-2.3.1-cp39-cp39-manylinux2014_aarch64.whl
ray-2.3.1-cp39-cp39-manylinux2014_x86_64.whl
ray-2.3.1-cp39-cp39-win_amd64.whl
ray-2.3.1+dbg-cp310-cp310-manylinux2014_x86_64.whl
ray-2.3.1+dbg-cp311-cp311-manylinux2014_x86_64.whl
ray-2.3.1+dbg-cp36-cp36m-manylinux2014_x86_64.whl
ray-2.3.1+dbg-cp37-cp37m-manylinux2014_x86_64.whl
ray-2.3.1+dbg-cp38-cp38-manylinux2014_x86_64.whl
ray-2.3.1+dbg-cp39-cp39-manylinux2014_x86_64.whl
```

**Comment by rickyyx** [comment]
> @rickyyx Good catch on the name change from `intel` to `x86-64` on 3.6 and 3.7. That means the `runtime_env` `"conda"` field won't work on 3.6 or 3.7 for in Ray 2.3.1 or on the nightly wheels without this fix. We should
> 
> 1. Cherry pick this for the release if possible, and
> 2. Find a way to run some kind of test in CI on all Python versions to catch this kind of name change before it happens (You noticed the name change manually and not because of some test failure, right?)

re1: Ah I see, so this fix is actually needed. 
re2: Yeah, seems we should somehow be able to catch this or not hardcode it in the future.

File: python/ray/_private/utils.py (hunk 1 above)
Comment (architkulkarni) [reply python/ray/_private/utils.py:1274]: This code just needs to reflect the current wheel file formats for the current Ray commit, no other implications outside of that

**Review by architkulkarni — APPROVED** [review]

**Comment by architkulkarni** [comment]
It might be too late to cherry pick a code change.  One idea is that this doesn't need to be cherry picked and we can just say it's not supported for 3.7 in a doc cherry pick, since according to the `conda` link you linked, `conda` itself no longer supports Python 3.7.

**Comment by rickyyx** [comment]
> It might be too late to cherry pick a code change. One idea is that this doesn't need to be cherry picked and we can just say it's not supported for 3.7 in a doc cherry pick, since according to the `conda` link you linked, `conda` itself no longer supports Python 3.7. (https://github.com/conda/conda/releases/tag/23.3.0)

Yeah, but I think the 3.7 deprecation only happens for the most recent conda release. People using older conda versions should still have this working I believe? 

Also, could you help us understand why do we need these wheels url with conda in the runtime env in high level?

**Comment by architkulkarni** [comment]
We need to download the Ray wheel into the conda environment because conda environments are isolated (there may be some way to get around this by having conda link to the existing Ray installation, but it wasn't straightforward.)

For releases, we could just write `'ray=={ray.__version__}` in the conda spec, but there would be no way to test this prior to the release going live on PyPI.  Furthermore, we want this to work for nightly wheels too, so we have to write the wheel URL in the conda spec.  So rather than use two different code paths for releases and nightly, we use the same wheel filenames in both cases.

**Comment by architkulkarni** [comment]
> Yeah, but I think the 3.7 deprecation only happens for the most recent conda release. People using older conda versions should still expect this to be working I believe?
Yeah, that's a good point.

**Comment by rickyyx** [comment]
I see - so sounds like we might have to hardcode it still? But maybe have tests test it explicitly to detect wheel name changes and valid wheel url available.

**Comment by rickyyx** [comment]
The failure tests fixed - but seems to run into some new failures: https://buildkite.com/ray-project/oss-ci-build-pr/builds/15059#0186ecd4-6e25-4b15-b5bc-dd4537d1116f

Rebased with master.

**Comment by rickyyx** [comment]
For 3.11 minimal test - I believe some dependencies the test depends currently are not supported yet. Skipping the test. https://github.com/ray-project/ray/issues/33415

**Pull request merged**